core/src/main/scala/com/acervera/osm4scala/utilities/CoordUtils.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023 Ángel Cervera Claudio
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.acervera.osm4scala.utilities
+
+import scala.math.BigDecimal.RoundingMode
+
+/**
+ * Utility to manage coordinates
+ */
+object CoordUtils {
+
+  private val COORDINATE_PRECISION = 7
+
+  /**
+   * Calculate coordinate applying offset, granularity and delta.
+   * The floating precision is 7 with Half Even rounding
+   * mode
+   *
+   * @param offSet
+   * @param delta
+   * @param currentValue
+   * @return
+   */
+  def decompressCoord(offSet: Long, delta: Long, currentValue: Double, granularity: Int): Double = {
```

Comment:
This changes are a good idea to increase precision, but are out of the scope of adding the new fields.
Could you move it out of this PR? Before to add it, we need to check if so precision is necessary for the precision level used in OSM and also the performance penalty that we are adding.

BTW, I'm not sure, but I think that the compiler is not going to optimize the code enough and the constants (like `(BigDecimal.valueOf(1E-9))`) are going to be created every time you execute the function.

So what do you think about remove these kind of changes from thi PR?

Reply:
i have removed them